developing/README.md

```diff
@@ -0,0 +1,196 @@
+## Setting up a full dev environment
+
+If you want to work on Proxstar using a 1:1 development setup, there are a couple things you're going to need
+
+- A machine you can SSH into, portforward from, and run Flask, Redis, and Docker on
+- At least one (1) Proxmox host running Proxmox >6.3
+- Docker
+- SSH portforwarding
+- A CSH account
+- An RTP (to tell you secrets)
+
+1. Configure your Proxmox node
+
+I would recommend setting up a development account on your Proxmox node. Name it anything. (Maybe `proxstartest`?). This is necessary to grab authentication tokens and the like. It should have the same permissions as `root@pam`. You can accomplish this by creating a group in `Datacenter > Permissions > Groups` and adding `Administrator` permissions to the group, then adding your user to the group. If you do this, then it's easy to enable/disable it for development. You should also generate an SSH key for the user.
+
+You will also have to set up a pool on your Proxmox node with your csh username. To do this, go into `Datacenter > Permissions > Pools > Create`.
+
+2. Set up your environment
+
+If you're trying to run this all on a VM without a graphical web browser, you can forward traffic to your computer using SSH.
+```
+ssh example@dev-server.csh.rit.edu -L 8000:localhost:8000
+```
+
+Clone down the repository, and create a Virtualenv to do your work in.
+```
+mkdir venv
+python3.8 -m venv venv
+source venv/bin/activate
+```
+
+Install required Python modules
+```
+pip install -r requirements.txt
+pip install click==7.1.2 
+pip install python-dotenv
+```
+Fill out the required fields in your config_local.py file. You might have to come back to this after you run the docker compose.
+```
+cp config.py config_local.py
+vim config_local.py
+```
+
+(Here's some advice on how to fill out your config file.)
```

Comment:
Maybe this config file should live in an example file instead of a codeblock? I've also seen `config.env.py` used...

Reply:
I think that would get crowded for someone who knows what they're doing. Also, and maybe we should start doing this, but no other CSH project really comments their config file. I think it's better to have some kind of a documentation page for that (GitHub wiki?)

Reply:
Fair enough